x-pack/solutions/security/plugins/security_solution/public/siem_migrations/common/types.ts

```diff
@@ -39,3 +42,36 @@ export interface FilterOptionsBase {
 export interface MigrationStats extends MigrationTaskStats {
   status: SiemMigrationTaskStatus; // use the native enum instead of the zod enum from the model
 }
+
+export enum SplunkDataInputStep {
+  Upload = 1,
+  Macros = 2,
+  Lookups = 3,
+  End = 10,
+}
+
+export interface Step<Props, C extends React.ComponentType<Props> = React.ComponentType<Props>> {
+  id: string;
+  Component: C;
+}
+
+interface MissingResourcesIndexed {
+  macros: string[];
+  lookups: string[];
+}
+
+export type OnMissingResourcesFetched = (missingResources: SiemMigrationResourceBase[]) => void;
+
+export enum MigrationSource {
+  SPLUNK = 'splunk',
+  QRADAR = 'qradar',
+}
+export interface UseMigrationStepsProps<T> {
```

Comment:
Can we simply call it `MigrationStepProps` or `SiemMigrationStepProps` ?

Reply:
Additionally, a `generic` may not be needed if you simply use type `MigrationStats` from line 42 above.